code.py

```diff
@@ -1,31 +1,36 @@
+from pathlib import Path
 from sqlalchemy import create_engine
 from time import time
 import pandas as pd
-import toml
-import os
+import os, glob, toml
 
-start = time()
+def sqlToParuet():
+    config = toml.load(open('config.toml'))
 
-config = toml.load(open('config.toml'))["config"]
+    if not os.path.exists('parquet'): os.mkdir('parquet')
 
-if not os.path.exists('parquet'): os.mkdir('parquet')
+    for database in config['databases']:
+        for species in database['species']:
 
-for elem in config:
-    if (elem["location"] == "local"):
-        pass
-    else:
-        db_connection_str = f'mysql+pymysql://{elem["DB_USER"]}@{elem["location"]}:{elem["port"]}/{elem["species"]}'
-        engine = create_engine(db_connection_str)
+            db_connection_str = f'mysql+pymysql://{database["DB_USER"]}@{database["location"]}:{database["port"]}/{species["DB_name"]}'
+            engine = create_engine(db_connection_str)
 
-        for table in elem["data"]["tables"]:
-            query = table["query"]
-            print(table["vars"])
-            df = pd.read_sql(query, con=engine, params=table["vars"])
+            for table in database['tables']:
+                df = pd.read_sql(table['query'], con=engine, params={'species_name': species['species_name']})
 
-            if not os.path.exists(f'parquet{os.sep}{elem["species"]}'):
-                os.mkdir(f'parquet{os.sep}{elem["species"]}')
+                directory = f'parquet{os.sep}data={table["table_name"]}{os.sep}species={species["species_name"]}'
```

Comment:
Yeah we need those parameters in the directory name so that we can query data using these params with the WHERE clause
For ex.
SELECT count(*) from ensembl_genome_data_parquet
WHERE data='gene' AND species='homo_sapiens'

Also partitioning our data like this makes sure Athena only crawls over the dir. which matches the criteria specified in the WHERE clause which means faster response time and cheaper queries
It is known as apache hive style partitions
Reference: https://docs.aws.amazon.com/athena/latest/ug/partitions.html

Reply:
Thanks for the clarification, I learned something new today :)

Reply:
@bilalebi Let me know your suggestions or if this sounds okay please mark this as resolved and approve the PR
I understand the issues with special chars in dir. name, it may required special handling in certain scenarios but in respect to your AWS setup this seemed like the best approach